tika-parsers/src/main/java/org/apache/tika/parser/crypto/TSDParser.java

```diff
@@ -0,0 +1,333 @@
+        List<TSDMetas> tsdMetasList = this.buildMetas(stream);
+        
+        Integer count = 1;
+        
+        for(TSDMetas tsdm: tsdMetasList) {
+            metadata.set(TSD_LOOP_LABEL + count + " - " + Metadata.CONTENT_TYPE, TSD_MIME_TYPE);
+            metadata.set(TSD_LOOP_LABEL + count + " - " + Metadata.DESCRIPTION, TSD_DESCRIPTION_VALUE);
+            metadata.set(TSD_LOOP_LABEL + count + " - " + this.TSD_PARSED_LABEL, tsdm.getParseBuiltStr());
+            metadata.set(TSD_LOOP_LABEL + count + " - " + this.TSD_PARSED_DATE, tsdm.getParsedDateStr() + " " + this.TSD_DATE_FORMAT);
+            metadata.set(TSD_LOOP_LABEL + count + " - " + this.TSD_DATE, tsdm.getEmitDateStr() + " " + this.TSD_DATE_FORMAT);
+            metadata.set(TSD_LOOP_LABEL + count + " - " + this.TSD_POLICY_ID, tsdm.getPolicyId());
+            metadata.set(TSD_LOOP_LABEL + count + " - " + this.TSD_SERIAL_NUMBER, tsdm.getSerialNumberFormatted());
+            metadata.set(TSD_LOOP_LABEL + count + " - " + this.TSD_TSA, tsdm.getTSAstr());
+            metadata.set(TSD_LOOP_LABEL + count + " - " + this.TSD_ALGORITHM, tsdm.getAlgorithmName());
+            count++;
+        }
+                
+        XHTMLContentHandler xhtml = new XHTMLContentHandler(handler, metadata);
+        xhtml.startDocument();
+        xhtml.endDocument();
+        
+     }
+    
+    private List<TSDMetas> buildMetas(InputStream stream) {
+        
+        List<TSDMetas> tsdMetasList = new ArrayList<TSDMetas>();
+        
+        try {
+             
+             CMSTimeStampedDataParser cmsTimeStampedData = new CMSTimeStampedDataParser(stream);
+             
+             TimeStampToken[] tokens = cmsTimeStampedData.getTimeStampTokens();
+             
+             for (int i=0; i < tokens.length; i++) {
+                 
+                 TSDMetas tsdMetas = new TSDMetas(true,
+                                                   tokens[i].getTimeStampInfo().getGenTime(),
+                                                   tokens[i].getTimeStampInfo().getPolicy().getId(),
+                                                   tokens[i].getTimeStampInfo().getSerialNumber(),
+                                                   tokens[i].getTimeStampInfo().getTsa(),
+                                                   tokens[i].getTimeStampInfo().getHashAlgorithm().getAlgorithm().getId());
+                 
+                 tsdMetasList.add(tsdMetas);
+             }
+             
+        } catch (Exception ex) {
+              tsdMetasList = new ArrayList<TSDMetas>();
+        }
+        
+        return tsdMetasList;
+    }
+    
+    private class TSDMetas {
+        
+        private final String DATE_FORMAT = "dd/MM/yyyy HH:mm:ss";
+        
+        private Boolean parseBuilt = false;
+        private Date emitDate = new Date();
+        private String policyId = "";
+        private BigInteger serialNumber = null;
+        private GeneralName TSA = null;
```

Comment:
It's better to use identifier starting with small letter for local and instance vars